pkg/apis/integreatly/v1alpha1/grafanadashboard_types.go

```diff
@@ -54,8 +44,7 @@ type GrafanaDashboard struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
-	Spec   GrafanaDashboardSpec   `json:"spec,omitempty"`
-	Status GrafanaDashboardStatus `json:"status,omitempty"`
```

Comment:
Adding an optional field is a non-breaking change, is removing one? I'm not sure :thinking: If not, then maybe this needs to be moved to a v1alpha2 or v1beta1?

Also, @pb82 any plans to use the operator-sdk to generate the CRDs from these types? The current ones look to probably have been manually created.

Reply:
Testing this on upgrade scenarios, both on standalone grafana deployments and even with integreatly there didn't seem to be any breaking changes, the dashboards updated to the expected state and removed status blocks, UID's were changed to our generated ones, and all folders were assigned correctly to their namespace-named folders.

Reply:
It's not a breaking change since the status present on existing dashboards will be ignored / removed. But in general our CRD versioning has room for improvement. @grdryn do you know how the operator would handle v1alpha1 dashboards when we upgrade the CRD and spec to v1alpha2? Once we create a new CRD version we can also use the operator sdk instead of writing them manually.

Reply:
Cool, so if it's not a breaking change, nothing to worry about :slightly_smiling_face: 

If you did want to make a breaking change in a new v1alpha2 CRD, I think there are a couple of options. Either:

* Don't worry about it, since you're still at alpha APIs, so there's no guarantee of backwards or long-term compatibility

or:

* Use a conversion webhook where you treat your v1alpha2 CRD similar to your v1alpha1 CRD at runtime. In this case I'd imagine you'd have a deprecation timeline of support for the old CRD, then in a future release you remove it and clean up. Users would have to migrate their v1alpha1 CRs to v1alpha2 equivalents before upgrading to the future version where v1alpha1 is gone completely. Example conversion webhook here: https://github.com/programming-kubernetes/pizza-crd/blob/master/pkg/webhook/conversion/convert.go#L61

or:

* It might be possible to have 2 different controllers, one for each version. At that point though, I'd consider calling the CRD something else.

Reply:
The conversion webhook solution sounds good. We can consider that when we add a breaking change. Thanks @grdryn !